### ISSUE TITLE
log

### ISSUE THREAD

**Issue by gmos** [amrog/pdi-capstone-budget-tracker#2]
Hi Aaron,

How are you going on the Newbie bites? 
I have seen you got stuck during Newbie Bite 26. Reading Errors and made and added a post on github and on the Pybit.es forum.  I also noticed that you were able to complete it. I have provided a walkthrough of the code on the forum and also a comment on the issue in github. 
Would you have the time to work on them a bit more? It would be nice to have them out of the way as soon as possible. Most fun and best learning would be on the project bit of the PDI.

Next the project.  From the project description we get that the the MVP version should implement the following:
Set Budget: User can set a monthly budget.
Add Expense: User can add a new expense.
View Expenses: User can view all expenses.
Total: User can see the total expenses.
Compare: User can compare total expenses with the set budget.
With the following implementation hints:
Define a variable to store the budget.
Define a list to store the expenses.
Display a menu with options (set budget, add expense, view expenses, total, compare, quit).
Based on user input, perform the following:
Set Budget: Ask for budget amount, store it in the budget variable.
Add Expense: Ask for expense details, add it to the expenses list.
View Expenses: Display all expenses.
Total: Calculate and display the total expenses.
Compare: Calculate and display the difference between total expenses and set budget. If expenses exceed the budget, indicate an overspending situation.
Quit: Break the loop and end the program.
For this first small Python you best keep the solution to one Python script file, the main.py file that already exists in your project repo.
As your application gets bigger you will cut it up and implement a proper project structure in your repo, but for now a single file is just fine.

This does not mean that there should not be a nice, understandable structure in your program.  You should not write one long script, but neatly cut up the code in separate functions.
As a starter I have expanded the main.py in your repo with some sample/boilerplate code.  I created a so called pull request (PR for short) for this:
The actual change was created on a separate git branch and is not part of your main branch yet.
In a separate list the PR was added referring to the change in the new branch and allowing you to check the proposed change, comment on it, reject or accept it.
Only after your acceptance will the proposed change be taken and made part of the main branch.
You could pull it in now or we could do this tomorrow in our call.  Remember that this change and the PR only exists on the GitHub copy of your repo.  After acceptance it will be added to the main branch, still on GitHub.
To get it locally on your machine you must update your local repository after the PR commit has been done on GitHub:
Make sure that the commit has been done on GitHub.  You can verify by displaying the updated main.py file in the GitHub browser.
Open a terminal and go to your project directory on your local machine. You can use Windows Terminal or the integrated terminal window in VSCode.
Then issue the git status command.  It should show you that your local repo is behind the remote one on GitHub by (at least) one commit.
Also check out the content of the main.py file, either in VSCode, or by using the type main.py command in the terminal.
Nuw run git pull in the terminal. This will pull the changes from the GitHub repo copy into your local copy and merge them onto the main branch.
Then issue the git status command again.  It should show you that your local repo is the same as the remote one.
Also check out the content of the main.py file, either in VSCode, or by using the type main.py command in the terminal. It should show the new content.
FInally the real exercise for tomorrow.  Try to work out a list of the functions you will need to write for your program.  These functions can be thought of as a network, where the arrows depict a "call" relationship.  The first node in the network is the main() function.
For now a list of functions and a short description of their functionality will do.  Something like:
name                  calls                              purpose
main                  greeting, menu, ...                main driver for the program. After greeting the user, runs a loop displaying the menu, get the menu choice and dispatches to the appropriate handler.
greeting                                                 greets the current user with a banner printed on screen.
menu                                                     displays the menu and returns with the user's choice.  Repeats asking for input until the choice is actually valid.
...

Until tomorrow.
